src/services/socket.tsx

```diff
@@ -3,40 +3,87 @@
 import { io } from 'socket.io-client'
 import { useEffect, useState } from 'react'
 
-type ListenerRes = {
-	messageServer: string
-	roomIdServer: string
-	userNameServer: string
+export type ListenerRes = {
+	message: string
+	// roomId: string
+	userName: string
+	timeStamp: string
 }
 
-export interface Event extends ListenerRes {
+type AllUsersPointsRes = {
+	message: ListenerRes[]
+	// roomId: string
+	userName: string
 	timeStamp: string
 }
 
 const socket = io()
 
 export function socketRoomEmitter(
 	emitterName: string,
-	message: string,
-	roomId: string,
+	message: string | ListenerRes[],
 	userName: string,
+	timeStamp: string,
+	roomId: string,
 ) {
-	socket.emit(emitterName, message, roomId, userName)
+	socket.emit(emitterName, message, userName, timeStamp, roomId)
+}
+
+export function useAllUsersPointsListener(listenerName: string) {
+	const [listenerRes, setListenerRes] = useState<AllUsersPointsRes>()
+	useEffect(() => {
+		function onListenerRes(
+			message: ListenerRes[],
+			// roomId: string,
+			userName: string,
+			timeStamp: string,
+		) {
+			setListenerRes({ message, userName, timeStamp })
+		}
+		socket.on(listenerName, onListenerRes)
+
+		return () => {
+			socket.off(listenerName, onListenerRes)
+		}
+	}, [listenerName])
+	return listenerRes
 }
 
 export function useSocketListener(listenerName: string) {
```

Comment:
todo: This `listenerName` (suggested above to be called `eventName` is not just a string. We need to create a type `Events` with all possible event names. This helps you spot typos or call events that no longer exist.

Reply:
I really don't get what you are suggesting. `listenerName` is the name of the Listener, so the name fits. It is only a string, so there is only one type. `listenerName` are things like "join-room" or "all-users-story-points".

Reply:
Ohhhhhh.......... Okay, I get that now. I've never used an enum. Okay, I'll make that change.

DONE!